samples/notebooks/DSL Static Type Checking.ipynb

```diff
@@ -0,0 +1,732 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# KubeFlow Pipeline DSL Static Type Checking\n",
+    "\n",
+    "In this notebook, we will demo: \n",
+    "\n",
+    "* Defining a KubeFlow pipeline with Python DSL\n",
+    "* Compile the pipeline with type checking\n",
+    "\n",
+    "Since this sample focuses on the DSL type checking, we will use components that are not runnable in the system but with various type checking scenarios.\n",
```

Comment:
Better to give some context on why user should use static type checking.

Reply:
I added the benefits of the static type checking.